### PR TITLE
HoC 2024 - Add Statsig events to track conversion

### PR DESCRIPTION
**Pull request by kelbyhawn**
Adds the following events to track conversion for Hour of Code 2024, and adds the `analytics_event_log_helper` on hourofcode.com.
- code.org
  - Visit to [code.org/hourofcode](http://code.org/hourofcode) 
    - `Code.org Hour of Code Landing Page Visited`
- hourofcode.com
  - Visit to http://hourofcode.com/learn 
    - `Hour of Code Activities Page Visited`
  - Click “Start” button on activity on [hourofcode.com/learn](http://hourofcode.com/learn)
    - `Hour of Code Activity Start Button Clicked`

## Links
Jira ticket: [ACQ-2312](https://codedotorg.atlassian.net/browse/ACQ-2312)

## Testing story
Tested locally

## Followup
As I start building out the other pages I'll add events if I see anything that would be good to track.

### CODE REVIEW

File: pegasus/sites.v3/code.org/public/hourofcode/index.haml [hunk 1]
```diff
@@ -96,3 +96,4 @@ theme: responsive_full_width
     %h2=hoc_s(:hoc_page_go_beyond_hoc)
     = view :hoc_page_beyond
 
+= view :analytics_event_log_helper, event_name: AnalyticsConstants::HOC_LANDING_PAGE_VISITED
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/public/hourofcode/index.haml:99]: HoC_landing_age

File: pegasus/sites.v3/hourofcode.com/public/learn/index.haml [hunk 2]
```diff
@@ -49,6 +49,8 @@ social:
 
 .clear{style: "clear: both"}
 
+= view :analytics_event_log_helper, event_name: AnalyticsConstants::HOC_ACTIVITIES_PAGE_VISITED
```
Comment (kelbyhawn) [inline pegasus/sites.v3/hourofcode.com/public/learn/index.haml:52]: Activites_page

File: apps/src/tutorialExplorer/tutorialDetail.jsx [hunk 3]
```diff
@@ -46,9 +49,19 @@ export default class TutorialDetail extends React.Component {
...
 
   startTutorialClicked = () => {
+    // Send event to Google Analytics
     const shortCode = this.props.item.short_code;
     ga('send', 'event', 'learn', 'start', shortCode);
     ga('send', 'event', 'learn', `start-${this.props.grade}`, shortCode);
+    // Send event to Statsig
+    const activityUrl = this.props.item.url;
+    analyticsReporter.sendEvent(
+      EVENTS.HOC_ACTIVITY_START_BUTTON_CLICKED,
```
Comment (kelbyhawn) [inline apps/src/tutorialExplorer/tutorialDetail.jsx:59]: Screenshot 2024-08-27 at 1 12 27 PM

File: apps/src/tutorialExplorer/tutorialDetail.jsx [hunk 4]
```diff
@@ -46,9 +49,19 @@ export default class TutorialDetail extends React.Component {
...
+      {
+        activityUrl: activityUrl,
```
Comment (kelbyhawn) [inline apps/src/tutorialExplorer/tutorialDetail.jsx:61]: @dmcavoy I added the activity url here, should I add anything else like the title?

Comment (dmcavoy) [reply apps/src/tutorialExplorer/tutorialDetail.jsx:61]: I think thats good to start with. Can't think of others that we won't get from statsig automatically

**Review by TurnerRiley — APPROVED** [review]
Yay this is great!

File: apps/src/sites/hourofcode.com/pages/views/analytics_event_log_helper.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import $ from 'jquery';
+
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+
+// Log page visits on document.ready.
+$(document).ready(() => {
+  const eventLogger = document.getElementById('analytics-event-log-helper');
+  const pageVisitEventName = eventLogger?.dataset?.eventName;
+  analyticsReporter.sendEvent(pageVisitEventName);
```
Comment (bethanyaconnor) [inline apps/src/sites/hourofcode.com/pages/views/analytics_event_log_helper.js:9]: @kelbyhawn @hannahbergam should this only be logging to statsig?

Comment (kelbyhawn) [reply apps/src/sites/hourofcode.com/pages/views/analytics_event_log_helper.js:9]: Yep! Updated this here: https://github.com/code-dot-org/code-dot-org/pull/60838 ([exact line](https://github.com/code-dot-org/code-dot-org/pull/60838/files#diff-e82046ae202d83dd15831248f036e7a8ab33a20f9ca4be4b67aa94e7175ae179R10))

**Pull request merged**